Graph-ASPNET-46-Snippets/Microsoft Graph ASPNET Snippets/Helpers/SDKHelper.cs

```diff
@@ -4,33 +4,41 @@
 */
 
 using Microsoft.Graph;
-using System.Net.Http.Headers;
-using System;
+using Microsoft_Graph_ASPNET_Snippets.TokenStorage;
+using Microsoft.Graph.Auth;
+using Microsoft.Identity.Client;
+using System.Web;
+using System.Configuration;
 
 namespace Microsoft_Graph_ASPNET_Snippets.Helpers
 {
     public class SDKHelper
     {
+        // Properties used to get and manage an access token.
+        private static string redirectUri = ConfigurationManager.AppSettings["ida:RedirectUri"];
+        private static string appId = ConfigurationManager.AppSettings["ida:AppId"];
+        private static string appSecret = ConfigurationManager.AppSettings["ida:AppSecret"];
+        private static string nonAdminScopes = ConfigurationManager.AppSettings["ida:NonAdminScopes"];
+        private static string adminScopes = ConfigurationManager.AppSettings["ida:AdminScopes"];
 
         // Get an authenticated Microsoft Graph Service client.
         public static GraphServiceClient GetAuthenticatedClient()
         {
-            GraphServiceClient graphClient = new GraphServiceClient(
-                new DelegateAuthenticationProvider(
-                    async (requestMessage) =>
-                    {
-                        string accessToken = await SampleAuthProvider.Instance.GetUserAccessTokenAsync();
+            if (!redirectUri.EndsWith("/")) redirectUri = redirectUri + "/";
+            bool? isAdmin = HttpContext.Current.Session["IsAdmin"] as bool?;
+            string allScopes = nonAdminScopes;
+            if (isAdmin.GetValueOrDefault())
+            {
+                allScopes += " " + adminScopes;
+            }
+            string[] scopes = allScopes.Split(new char[] { ' ' });
 
-                        // Append the access token to the request.
-                        requestMessage.Headers.Authorization = new AuthenticationHeaderValue("bearer", accessToken);
+            HttpContextBase context = HttpContext.Current.GetOwinContext().Environment["System.Web.HttpContextBase"] as HttpContextBase;
+            SessionTokenCacheProvider sessionTokenCacheProvider = new SessionTokenCacheProvider(context);
 
-                        // Get event times in the current time zone.
-                        requestMessage.Headers.Add("Prefer", "outlook.timezone=\"" + TimeZoneInfo.Local.Id + "\"");
```

Comment:
Did this code get moved somewhere else?

Reply:
@darrelmiller it's actually omitted. We have two approaches to add the prefer header :
- Use `HttpProvider` and pass custom `HttpMessageHandler` that set's the header.
- Add the header as an option to our request builders i.e. `me.Request(headerOption).GetAsync()` for all requests.

Reply:
@pwombwa Peter, can you update the PR to also update the version of MSAL from 1.1.x to 2.7.1

@darrelmiller Besides the update to MSAL, is there more work to do here or can we move ahead with the PR once that's done?